### DISCUSSION TITLE
How to export arxiv ID in publication field

### DISCUSSION THREAD

**Discussion by mmwoodmath** [retorquere/zotero-better-bibtex#2355] [Q&A] [answered]
Zotero used to file arxiv papers as journal articles, and put something like "arXiv:1701.02838 [math]" in the publication field.  This was perfect for my use case.  The "[math]" is not important to me.

Now Zotero classifies arxiv papers as preprints, but thanks to some helpful advice from emilianoeheyns, I used the following code

```
if (Translator.BetterBibTeX && zotero.itemType === 'preprint') {
  tex.entrytype = 'article'
}
```

to export them as articles in my BibTex file.  

My question now is how to get the arxiv ID to export into the journal field in my BibTex file, so as to reproduce to old behavior in terms of my BibTex file.

The most recent downloads from the arxiv put the arxiv ID in a Zotero field "Archive ID".

Thanks!

**Comment by retorquere** [comment]
If you can get me a debug log (right-click on the entry, better bibtex, send debug log) I can get you a tailored answer.

(I'm emilianoeheyns on the forums BTW)

**Reply by mmwoodmath** [reply]
I submitted a debug log:
F33PFLD2-refs-euc

Thank you!

**Comment by retorquere** [comment, marked as answer]
Can you try

```
if (Translator.BetterBibTeX && zotero.itemType === 'preprint') {
  tex.entrytype = 'article'
  if (zotero.arXiv) tex.add({ name: 'journal', value: zotero.arXiv.id }) 
}
```

**Reply by mmwoodmath** [reply]
Thank you!  I wanted "arXiv:" to appear so I changed it slightly to:

```
if (Translator.BetterBibTeX && zotero.itemType === 'preprint') {
  tex.entrytype = 'article'
  if (zotero.arXiv) tex.add({ name: 'journal', value: 'arXiv:'+zotero.arXiv.id }) 
}
```

and it does what I want.

**Reply by retorquere** [reply]
@imxtx if you want help, please state your actual problem rather than "not work", and send a debug log.

**Reply by imxtx** [reply]
Make the exported same as the Google Scholar:
```
if (Translator.BetterBibTeX && zotero.itemType === 'preprint') {
  tex.entrytype = 'article'
  if (zotero.arXiv) {
    tex.add({ name: 'journal', value: 'arXiv preprint arXiv:'+zotero.arXiv.id })
    tex.remove('number')
  }
}
```

**Reply by imxtx** [reply]
> @imxtx if you want help, please state your actual problem rather than "not work", and send a debug log.

Thank you. I figured it out. I should copy BibTeX instead of BibLaTeX.

**Comment by shunzh** [comment]
I hoped I have found this thread earlier, but the code I use is
```
if (Translator.BetterTeX && zotero.arXiv) {
  tex.add({ name: 'note', value: "arXiv:" + zotero.arXiv.id + " [" + zotero.arXiv.category + "]" });
}
```
This is to recover `arXiv:{id} [{categroy}]`, the standard arxiv citation format.

However, I don't understand why we need to do this extra step to make arXiv citations displayed correctly. Without Better Bibtex, the citations were shown correctly in Zotero exports.

**Reply by retorquere** [reply]
If you want a change, open an issue.

**Reply by retorquere** [reply]
(I'm not going to stick the arXiv ID in a note by default)
